studio/data/database/indexes-query.ts

```diff
@@ -0,0 +1,55 @@
+import { UseQueryOptions } from '@tanstack/react-query'
+import { ExecuteSqlData, useExecuteSqlQuery } from '../sql/execute-sql-query'
+import { ResponseError } from 'types'
+
+// [Joshen] Future refactor to move to pg-meta once available
+
+export type DatabaseIndex = {
+  name: string
+  schema: string
+  table: string
+  definition: string
+  enabled: boolean
+}
+
+export const getIndexesQuery = ({ schema }: { schema: string }) => {
+  const sql = /* SQL */ `
+SELECT schemaname as "schema",
+  tablename as "table",
+  indexname as "name",
+  indexdef as "definition",
+  indisvalid as "enabled"
+FROM pg_indexes a
+  LEFT JOIN pg_index b ON b.indexrelid = a.indexname::regclass
+WHERE schemaname = '${schema}';
```

Comment:
As mentioned in your comment, it could be nice to have this in pg-meta – do you think others will find it useful @soedirgo?

Reply:
yup there's currently a tracked ticket in [here](https://www.notion.so/pg-meta-retrieve-indexes-53bf554b013b401bb14bd88f62349b22) RE adding pg-meta support for indexes

FE's SQL implementation is just to fill the gaps for now (and gives us the nice "open into SQL editor" UX too haha)